From aee9ede2e75b2a5b7ca374eb2a6041e236c13a86 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Tue, 10 Jan 2017 14:40:07 +0000
Subject: [PATCH] Resolves: rhbz#1409634 crash on close storing clipboard of
 database table

Change-Id: I8e692314b4ef278f7881780d51bcecb60ca80c3f
(cherry picked from commit 6c2725038b627535ba97ead89dadb96e0ff69c3c)
---
 dbaccess/source/ui/misc/TokenWriter.cxx | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/dbaccess/source/ui/misc/TokenWriter.cxx b/dbaccess/source/ui/misc/TokenWriter.cxx
index 98a3312..cac82ba 100644
--- a/dbaccess/source/ui/misc/TokenWriter.cxx
+++ b/dbaccess/source/ui/misc/TokenWriter.cxx
@@ -934,8 +934,7 @@ void OHTMLImportExport::WriteCell( sal_Int32 nFormat, sal_Int32 nWidthPixel, sal
     aStrTD = aStrTD + "=";
     aStrTD = aStrTD + pChar;
 
-    Reference< css::util::XNumberFormatsSupplier >  xSupplier = m_xFormatter->getNumberFormatsSupplier();
-    SvNumberFormatsSupplierObj* pSupplierImpl = SvNumberFormatsSupplierObj::getImplementation( xSupplier );
+    SvNumberFormatsSupplierObj* pSupplierImpl = m_xFormatter.is() ? SvNumberFormatsSupplierObj::getImplementation(m_xFormatter->getNumberFormatsSupplier()) : nullptr;
     SvNumberFormatter* pFormatter = pSupplierImpl ? pSupplierImpl->GetNumberFormatter() : nullptr;
     if(pFormatter)
     {
-- 
2.9.3

